application/controllers/jwt.js

```diff
@@ -7,7 +7,8 @@ Atm we save the userid because just the delete operation for a user is restricte
 const jwt = require('jsonwebtoken'),
   config = require('../configuration');
 
-module.exports = {
+const self = module.exports = {
```

Comment:
Why have you added `const self = ...`? I've done a quick google search about it, but found no reason.

Reply:
I need to call `createToken` inside the same file, I find it better than using `module.exports.createToken`. I use it in deck service already. BUT I may have to change it, as I just saw this: https://developer.mozilla.org/en-US/docs/Web/API/Window/self which basically means self is an alias for the global object :(

Reply:
hmmm but not for nodejs... also I've seen this code style here and there, that's why I picked it up...

Reply:
Ah okay. I'm not a 100% sure but you should also be able to use it by `this.xyz();`, or am I wrong?
+ You're not using `createToken()` from another module, so you must not export it at all.

Reply:
Roy is right. JWT will only include user data (in this service and perhaps in general) thus jwt.js should only provide one function: createToken

Reply:
@rmeissn `this` would not work, as it depends where it's bound. For this particular example, `this` would refer to the global object, not the module. I could however override `this`, e.g. with ```
let createToken = jwt.createToken.bind(anyObjectHere);
createToken(user);
```
which would break the code. Using `const self` which is a local (private) variable in the file makes sure `self` is always the module exports.